nano/rpc_test/rpc.cpp

```diff
@@ -2627,16 +2627,124 @@ TEST (rpc, krai_from_raw)
 	ASSERT_EQ ("1", response1.get<std::string> ("amount"));
 }
 
-TEST (rpc, nano_to_raw)
+TEST (rpc, nano_to_raw_one_nano)
 {
 	nano::system system;
 	auto node1 = add_ipc_enabled_node (system);
 	auto const rpc_ctx = add_rpc (system, node1);
 	boost::property_tree::ptree request1;
 	request1.put ("action", "nano_to_raw");
 	request1.put ("amount", "1");
-	auto response1 (wait_response (system, rpc_ctx, request1));
-	ASSERT_EQ (nano::Mxrb_ratio.convert_to<std::string> (), response1.get<std::string> ("amount"));
+	auto response (wait_response (system, rpc_ctx, request1));
+	ASSERT_EQ (nano::Mxrb_ratio.convert_to<std::string> (), response.get<std::string> ("amount"));
+}
+
+TEST (rpc, nano_to_raw_zero_nano)
+{
+	nano::system system;
+	auto node1 = add_ipc_enabled_node (system);
+	auto const rpc_ctx = add_rpc (system, node1);
+	boost::property_tree::ptree request1;
+	request1.put ("action", "nano_to_raw");
+	request1.put ("amount", "0");
+	auto response (wait_response (system, rpc_ctx, request1));
+	ASSERT_EQ ("0", response.get<std::string> ("amount"));
+}
+
+TEST (rpc, nano_to_raw_one_tenth_nano)
+{
+	nano::system system;
+	auto node1 = add_ipc_enabled_node (system);
+	auto const rpc_ctx = add_rpc (system, node1);
+	boost::property_tree::ptree request1;
+	request1.put ("action", "nano_to_raw");
+	request1.put ("amount", "0.1");
+	auto response (wait_response (system, rpc_ctx, request1));
+	ASSERT_EQ ((nano::Mxrb_ratio / 10).convert_to<std::string> (), response.get<std::string> ("amount"));
+}
+
+TEST (rpc, nano_to_raw_one_raw)
+{
+	nano::system system;
+	auto node1 = add_ipc_enabled_node (system);
+	auto const rpc_ctx = add_rpc (system, node1);
+	boost::property_tree::ptree request1;
+	request1.put ("action", "nano_to_raw");
+	request1.put ("amount", "0.000000000000000000000000000001");
+	auto response (wait_response (system, rpc_ctx, request1));
+	ASSERT_EQ ("1", response.get<std::string> ("amount"));
+}
+
+TEST (rpc, nano_to_raw_value_less_than_one_raw)
```

Comment:
If the whole point of this feature is to ensure perfect accurancy in the nano range then this against that rule. We are losing accuracy here. I think this should return an error.

Reply:
I agree. I've added a new error type "invalid_amount_loss_of_precision" for this case